repos/system_upgrade/common/libraries/rpms.py

```diff
@@ -52,15 +55,25 @@ def has_package(model, package_name, arch=None, version=None, release=None, cont
     """
     if not (isinstance(model, type) and issubclass(model, InstalledRPM)):
         return False
-    keys = ['name']
-    if arch:
-        keys.append('arch')
-    if version:
-        keys.append('version')
-    if release:
-        keys.append('release')
+    return tuple(attributes) in package_data_for(model,package_name,arch,version,release,context)
```

Comment:
`attributes` is nonsensical in the new `has_package`. Should this now just pass through what `package_data_for` returns?